### PR TITLE
Check XProckLock return values

### PR DESCRIPTION
**Pull request by p-steuer**
Resolve issue https://github.com/opencryptoki/opencryptoki/issues/35 .

### CODE REVIEW

File: usr/lib/common/loadsave.c [hunk 1]
```diff
@@ -452,7 +458,14 @@ CK_RV load_token_data(STDLL_TokData_t *tokdata, CK_SLOT_ID slot_id)
...
 
 out_unlock:
-    XProcUnLock(tokdata);
+    if (rc == CKR_OK) {
+        rc = XProcUnLock(tokdata);
+	if (rc != CKR_OK)
+            TRACE_ERROR("Failed to release Process Lock.\n");
+    } else {
+        /* return error that occurred first */
+	XProcUnLock(tokdata);
```
Comment (ifranzki) [inline usr/lib/common/loadsave.c]: You have some tabs here instead of spaces (line 463 and 467)

File: usr/lib/common/loadsave.c [hunk 2]
```diff
@@ -502,7 +515,14 @@ CK_RV save_token_data(STDLL_TokData_t *tokdata, CK_SLOT_ID slot_id)
...
 
 done:
-    XProcUnLock(tokdata);
+    if (rc == CKR_OK) {
+        rc = XProcUnLock(tokdata);
+	if (rc != CKR_OK)
+            TRACE_ERROR("Failed to release Process Lock.\n");
+    } else {
+        /* return error that occurred first */
+	XProcUnLock(tokdata);
```
Comment (ifranzki) [inline usr/lib/common/loadsave.c]: You have some tabs here instead of spaces (lines 520 and 524)

File: usr/lib/common/obj_mgr.c [hunk 3]
```diff
@@ -131,62 +131,64 @@ CK_RV object_mgr_add(STDLL_TokData_t *tokdata,
...
+            // TODO: handle error, check if rc is a valid per spec
+            XProcUnLock(tokdata);
+            goto done;
         }
 
+        rc = XProcUnLock(tokdata);
+        if (rc != CKR_OK) {
+            TRACE_ERROR("Failed to release Process Lock.\n");
+            goto done;
+	}
```
Comment (ifranzki) [inline usr/lib/common/obj_mgr.c]: You have some tabs here instead of spaces

File: usr/lib/common/utility.c [hunk 4]
```diff
@@ -757,28 +759,34 @@ CK_RV attach_shm(STDLL_TokData_t *tokdata, CK_SLOT_ID slot_id)
...
 
-    XProcLock(tokdata);
+    rc = XProcLock(tokdata);
+    if (rc != CKR_OK)
+        goto err;
 
     if (sm_close((void *) tokdata->global_shm, 0)) {
         TRACE_DEVEL("sm_close failed.\n");
         rc = CKR_FUNCTION_FAILED;
+	goto err;
```
Comment (ifranzki) [inline usr/lib/common/utility.c]: You have some tabs here instead of spaces

File: usr/lib/icsf_stdll/icsf_specific.c [hunk 5]
```diff
@@ -1625,10 +1695,20 @@ CK_RV icsftok_generate_key_pair(STDLL_TokData_t * tokdata, SESSION * session,
     }
 
     /* Copy token name from shared memory */
-    XProcLock(tokdata);
+    rc = XProcLock(tokdata);
+    if (rc != CKR_OK) {
+        TRACE_ERROR("Failed to get process lock.\n");
+	goto done;
```
Comment (ifranzki) [inline usr/lib/icsf_stdll/icsf_specific.c]: You have some tabs here instead of spaces

File: usr/lib/icsf_stdll/icsf_specific.c [hunk 6]
```diff
@@ -1625,10 +1695,20 @@ CK_RV icsftok_generate_key_pair(STDLL_TokData_t * tokdata, SESSION * session,
...
+    }
+
     memcpy(token_name, tokdata->nv_token_data->token_info.label,
            sizeof(token_name));
-    XProcUnLock(tokdata);
+
+    rc = XProcUnLock(tokdata);
+    if (rc != CKR_OK) {
+        TRACE_ERROR("Failed to release process lock.\n");
+	goto done;
```
Comment (ifranzki) [inline usr/lib/icsf_stdll/icsf_specific.c]: You have some tabs here instead of spaces

File: usr/lib/icsf_stdll/icsf_specific.c [hunk 7]
```diff
@@ -1710,10 +1790,20 @@ CK_RV icsftok_generate_key(STDLL_TokData_t * tokdata, SESSION * session,
         goto done;
 
     /* Copy token name from shared memory */
-    XProcLock(tokdata);
+    rc = XProcLock(tokdata);
+    if (rc != CKR_OK) {
+        TRACE_ERROR("Failed to get process lock.\n");
+	goto done;
```
Comment (ifranzki) [inline usr/lib/icsf_stdll/icsf_specific.c]: You have some tabs here instead of spaces

File: usr/lib/icsf_stdll/icsf_specific.c [hunk 8]
```diff
@@ -1710,10 +1790,20 @@ CK_RV icsftok_generate_key(STDLL_TokData_t * tokdata, SESSION * session,
...
+    }
+
     memcpy(token_name, tokdata->nv_token_data->token_info.label,
            sizeof(token_name));
-    XProcUnLock(tokdata);
+
+    rc = XProcUnLock(tokdata);
+    if (rc != CKR_OK) {
+        TRACE_ERROR("Failed to release process lock.\n");
+	goto done;
```
Comment (ifranzki) [inline usr/lib/icsf_stdll/icsf_specific.c]: You have some tabs here instead of spaces

File: usr/lib/icsf_stdll/icsf_specific.c [hunk 9]
```diff
@@ -3047,10 +3137,20 @@ CK_RV icsftok_find_objects_init(STDLL_TokData_t * tokdata, SESSION * sess,
     /* Prepare to query ICSF for list objects
      * Copy token name from shared memory
      */
-    XProcLock(tokdata);
+    rc = XProcLock(tokdata);
+    if (rc != CKR_OK) {
+        TRACE_ERROR("Failed to get process lock.\n");
+	return rc;
```
Comment (ifranzki) [inline usr/lib/icsf_stdll/icsf_specific.c]: You have some tabs here instead of spaces

File: usr/lib/icsf_stdll/icsf_specific.c [hunk 10]
```diff
@@ -3047,10 +3137,20 @@ CK_RV icsftok_find_objects_init(STDLL_TokData_t * tokdata, SESSION * sess,
...
+    }
+
     memcpy(token_name, tokdata->nv_token_data->token_info.label,
            sizeof(token_name));
-    XProcUnLock(tokdata);
+
+    rc = XProcUnLock(tokdata);
+    if (rc != CKR_OK) {
+        TRACE_ERROR("Failed to release process lock.\n");
+	return rc;
```
Comment (ifranzki) [inline usr/lib/icsf_stdll/icsf_specific.c]: You have some tabs here instead of spaces

File: usr/lib/icsf_stdll/icsf_specific.c [hunk 11]
```diff
@@ -4554,10 +4654,20 @@ CK_RV icsftok_derive_key(STDLL_TokData_t * tokdata, SESSION * session,
         return rc;
 
     /* Copy token name from shared memory */
-    XProcLock(tokdata);
+    rc = XProcLock(tokdata);
+    if (rc != CKR_OK) {
+        TRACE_ERROR("Failed to get process lock.\n");
+	return rc;
```
Comment (ifranzki) [inline usr/lib/icsf_stdll/icsf_specific.c]: You have some tabs here instead of spaces

File: usr/lib/icsf_stdll/icsf_specific.c [hunk 12]
```diff
@@ -4554,10 +4654,20 @@ CK_RV icsftok_derive_key(STDLL_TokData_t * tokdata, SESSION * session,
...
+    }
+
     memcpy(token_name, tokdata->nv_token_data->token_info.label,
            sizeof(token_name));
-    XProcUnLock(tokdata);
+
+    rc = XProcUnLock(tokdata);
+    if (rc != CKR_OK) {
+        TRACE_ERROR("Failed to release process lock.\n");
+	return rc;
```
Comment (ifranzki) [inline usr/lib/icsf_stdll/icsf_specific.c]: You have some tabs here instead of spaces

File: usr/lib/tpm_stdll/tpm_specific.c [hunk 13]
```diff
@@ -1809,10 +1809,22 @@ CK_RV token_specific_login(STDLL_TokData_t * tokdata, SESSION * sess,
         }
 
         rc = load_private_token_objects(tokdata);
+	if (rc != CKR_OK)
```
Comment (ifranzki) [inline usr/lib/tpm_stdll/tpm_specific.c]: You have some tabs here instead of spaces

**Comment by p-steuer** [comment]
Fixed the auto-ident stuff. Will be merged when travis is done.

**Pull request merged**